packages/typespec-ts/src/modular/buildCodeModel.ts

```diff
@@ -1030,9 +1027,7 @@ function emitModel(
   }
   const effectiveName = getEffectiveSchemaType(context.program, type).name;
   const overridedModelName =
-    getProjectedName(context.program, type, "javascript") ??
-    getProjectedName(context.program, type, "client") ??
-    getFriendlyName(context.program, type);
+    getLibraryName(context, type) ?? getFriendlyName(context.program, type);
```

Comment:
getLibraryName will return the `@clientName` value first, then getProjectedName for each emitter.